lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2971,35 +3034,41 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
       SV = transValue(II->getOperand(0), BB);
     }
 
-    GetElementPtrInst *GEP = cast<GetElementPtrInst>(II->getArgOperand(1));
-    Constant *C = cast<Constant>(GEP->getOperand(0));
-    StringRef AnnotationString;
-    getConstantStringInfo(C, AnnotationString);
+    std::string AnnotationString;
+    if (GetElementPtrInst *GEP =
+            dyn_cast<GetElementPtrInst>(II->getArgOperand(1))) {
```

Comment:
May there be a case when var_annotation has non GEP 2nd argument? Should we try to process 5th argument if we couldn't translate 2nd?

Reply:
>> May there be a case when var_annotation has non GEP 2nd argument?

If I understand correctly, it's not a real-life scenario. LangRef says, that var/ptr/annotation should always have 2nd and 3rd arguments to be pointers to a global string and we always accesses to a global string via GEP instruction aka via a pointer to [0] element with a specified memory size.